src/librustc_typeck/variance/constraints.rs

```diff
@@ -315,8 +315,8 @@ impl<'a, 'tcx> ConstraintContext<'a, 'tcx> {
                 self.add_constraints_from_region(current, r, contra);
 
                 if let Some(poly_trait_ref) = data.principal() {
-                    let poly_trait_ref =
-                        poly_trait_ref.with_self_ty(self.tcx(), self.tcx().types.err);
+                    let poly_trait_ref = poly_trait_ref
+                        .with_self_ty(self.tcx(), self.tcx().types.trait_object_dummy_self);
                     self.add_constraints_from_trait_ref(
```

Comment:
Oh, I see, `add_constraints_from_trait_ref` is trivial, you can just inline it, and then you'll be able to use `poly_trait_ref.substs` directly instead of there being a `with_self_ty` call.

Reply:
Done in https://github.com/rust-lang/rust/pull/70870/commits/efb8f07e888105d3dddc4baf612d551ef0805475